pennylane/devices/default_qubit.py

```diff
@@ -186,6 +186,17 @@ def __init__(
             "Toffoli": self._apply_toffoli,
         }
 
+    @property
+    def stopping_condition(self):
+        def accepts_obj(obj):
+            if getattr(obj, "has_matrix", False):
+                # pow operations dont work with backprop or adjoint without decomposition
+                # use class name string so we don't need to use isisntance check
+                return not (obj.__class__.__name__ == "Pow" and qml.operation.is_trainable(obj))
+            return obj.name in self.observables.union(self.operations)
```

Comment:
@albi3ro I realize that this differs from the inherited version:
```python
    @property
    def stopping_condition(self):
        """.BooleanFn: Returns the stopping condition for the device. The returned
        function accepts a queuable object (including a PennyLane operation
        and observable) and returns ``True`` if supported by the device."""
        return qml.BooleanFn(
            lambda obj: not isinstance(obj, qml.tape.QuantumTape)
            and self.supports_operation(obj.name)
        )
```

Should we call `super()` here? Might be safer, in case someone changes the default in the `Device` class

Reply:
If anything, we should change `Device.stopping_condition`. It's called on both operations and measurement processes in `tape.expand`, but the `Device` version only pays attention to operations.  This is highly problematic, and I am unsure why it wasn't causing problems before.

Going forward, we should have separate stopping conditions for operations and measurement processes, but that is out of scope for this PR.